report/report.py

```diff
@@ -0,0 +1,99 @@
+from sisyphus import *
+
+import getpass
+import gzip
+from typing import Dict, Union, Callable, Optional
+import pprint
+
+_Report_Type = Dict[str, Union[tk.AbstractPath, str]]
+
+
+class GenerateReportStringJob(Job):
+    """
+    Job to generate and output a report string
+    """
+
+    def __init__(
+        self,
+        report_values: _Report_Type,
+        report_template: Optional[Callable[[_Report_Type], str]] = None,
+        compress: bool = True,
+    ):
+        """
+
+        :param report_values:
+        :param report_template:
+        :param compress:
+        """
+        self.report_values = report_values
+        self.report_template = report_template
+        self.compress = compress
+
+        self.out_report = self.output_path("report.gz" if self.compress else "report")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+
+        if self.report_template:
+            report = self.report_template(self.report_values)
+        elif callable(self.report_values):
+            report = str(self.report_values())
+        else:
+            report = pprint.pformat(self.report_values, width=140)
```

Comment:
I am not sure about this if...elif...else clause.

why should the `report_values` be callable?? it is only of type `Dict`?!

the check on `report_template` could be restricted to None or even a callable..

Reply:
The typing was wrong. So this can be callable (similar to how sisyphus register report has it as a possibilty) , for example some function that returns some string (interpretable) output. You cannot pass the string directly because you dont want to generate the string at graph construction different reasons.
This is quite close to just passing a format function but since sisyphus supported it we decided to also do this here.